moved/src/state_actor.rs

```diff
@@ -450,6 +460,36 @@ impl<
         (outcome, receipts)
     }
 
+    fn simulate_transaction(&self, request: TransactionRequest) -> TransactionExecutionOutcome {
+        let mut tx = NormalizedEthTransaction::from(request.clone());
+        if let Some(from) = request.from {
+            tx.nonce = quick_get_nonce(&from.to_move_address(), self.state.resolver());
+        }
+        let tx = NormalizedExtendedTxEnvelope::Canonical(tx);
+
+        let block_header = HeaderForExecution {
+            number: self.height + 1,
+            timestamp: SystemTime::now()
+                .duration_since(UNIX_EPOCH)
+                .expect("Should get current time")
+                .as_secs(),
+            prev_randao: B256::random(),
+        };
+
+        match execute_transaction(
+            &tx,
+            &B256::random(),
+            self.state.resolver(),
+            &self.genesis_config,
+            0,
+            &(),
+            block_header,
+        ) {
+            Ok(outcome) => outcome,
+            Err(_) => unreachable!("User errors are handled in execution"),
```

Comment:
Errors other than user errors are possible here. For example, if the user provides their own gas limit then they could theoretically trigger `InvalidTransaction(InvalidTransactionCause::InsufficientIntrinsicGas)`.

I think this function should return a result and the `Err` case is handled in the JSON RPC by returning an error in the response.

Reply:
Returning the error now, but as it is.
Should we respond with a standard message and ignore the actual internal error message?